pkg/client/common.go

```diff
@@ -0,0 +1,82 @@
+package client
+
+import (
+	"fmt"
+
+	apistatus "github.com/nspcc-dev/neofs-api-go/pkg/client/status"
+	"github.com/nspcc-dev/neofs-api-go/v2/session"
+	"github.com/nspcc-dev/neofs-api-go/v2/signature"
+)
+
+// common interface of resulting structures with API status.
+type resCommon interface {
+	setStatus(apistatus.Status)
+}
+
+type statusRes struct {
+	st apistatus.Status
+}
+
+// setStatus implements resCommon interface method.
+func (x *statusRes) setStatus(st apistatus.Status) {
+	x.st = st
+}
+
+func (x statusRes) Status() apistatus.Status {
+	return x.st
+}
+
+type responseV2 interface {
+	GetMetaHeader() *session.ResponseMetaHeader
+	GetVerificationHeader() *session.ResponseVerificationHeader
+}
+
+// checks response signature and write client error if it is not correct (in this case returns true).
+func isInvalidSignatureV2(res *processResponseV2Res, resp responseV2) bool {
+	err := signature.VerifyServiceMessage(resp)
+
+	isErr := err != nil
+	if isErr {
+		res.cliErr = fmt.Errorf("invalid response signature: %w", err)
+	}
+
+	return isErr
+}
+
+type processResponseV2Prm struct {
+	callOpts *callOptions
+
+	resp responseV2
+}
+
+type processResponseV2Res struct {
+	statusRes resCommon
+
+	cliErr error
+}
+
+// performs common actions of response processing and writes any problem as a result status or client error
+// (in both cases returns true).
```

Comment:
returns true if there is an error? why not false? it is not clear like with `isInvalidSignatureV2` naming

Reply:
Because in all usages we place it in `if processResponseV2 { return }` statement, so we can skip `!` operator. Same is for `isInvalidSignatureV2`.

Reply:
but in `isInvalidSignatureV2` case it is clear with naming, and `processResponseV2` can confuse a little i think

Reply:
`processResponseV2` is not a verifier - it does some actions, and result just tells continue the processing or not.